Starwatch.Core/API/Rest/Route/AnnouncementRoute.cs

```diff
@@ -0,0 +1,387 @@
+                {
+                    // probably unnecessary
+                    id = 0;
+                }
+
+                if (id < 0)
+                {
+                    return new RestResponse(RestStatus.BadRequest, ID_MUST_BE_POSITIVE);
+                }
+
+                if (announcements.Length > id)
+                {
+                    return new RestResponse(RestStatus.OK, announcements[id]);
+                }
+
+                return new RestResponse(RestStatus.ResourceNotFound, MISSING_RESOURCE);
+            }
+            else
+            {
+                // Returns all announcements
+                if (announcements.Length == 1)
+                {
+                    return new RestResponse(RestStatus.OK, "There is 1 announcement.", 1);
+                }
+
+                return new RestResponse(RestStatus.OK, $"There are {announcements.Length} announcements.", announcements);
+            }
+        }
+
+        /// <summary>
+        /// Deletes the account
+        /// </summary>
+        /// <param name="query"></param>
+        /// <returns></returns>
+        public override RestResponse OnDelete(Query query)
+        {
+            // Grab the configurator from the monitor 
+            AnnouncementMonitor mon = null;
+            bool hasMon = Starbound.TryGetMonitor(out mon);
+
+            if (!hasMon)
+            {
+                return new RestResponse(RestStatus.InternalError, NO_MONITOR);
+            }
+
+            if (query.ContainsKey("id"))
+            {
+                int id = 0;
+                bool ok = int.TryParse(query["id"], out id);
+
+                if (!ok)
+                {
+                    return new RestResponse(RestStatus.BadRequest, ID_MUST_BE_INTEGER);
+                }
+
+                if (id < 0)
+                {
+                    return new RestResponse(RestStatus.BadRequest, ID_MUST_BE_POSITIVE);
+                }
+
+                if (!mon.Lock())
+                {
+                    return new RestResponse(RestStatus.InternalError, RESOURCE_LOCKED);
+                }
+
+                if (mon.Announcements.Length > id)
+                {
+                    try
+                    {
+                        AnnouncementMonitor.Announcement target = mon.Announcements[id];
+
+                        mon.Announcements = mon.Announcements.Where((s, i) => i != id).ToArray();
+                        mon.Configuration.SetKey("announcements", mon.Announcements);
+                        mon.Configuration.Save();
+
+                        mon.Unlock();
+                        mon.Reload();
+
+                        return new RestResponse(RestStatus.OK, $"Deleted announcement #{id}", target);
+                    }
+                    catch (Exception ex)
+                    {
+                        mon.Unlock();
+                        return new RestResponse(RestStatus.InternalError, "Failed to save config: " + ex.Message);
```

Comment:
~~Either `throw` and let the REST handler... handle it ( ba-dum-tish ) or return false.~~

edit: i forgot how this structure works. Ignore this comment it's good.

Reply:
``Throw``ing is scary lol. Starwatch has random crashes during REST calls and I haven't tracked it down yet since it doesn't appear to happen on the local test server.